### ISSUE TITLE
Slowing the rate of skill gain

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#67580]
### Is your feature request related to a problem? Please describe.

Skill gain in this game is presently broken. This is likely to be controversial in some circles, but we know it's true. For example, it takes 3 minutes in game to advance to skill level 1 from untrained, and in general early skill levels are gated somewhat by the time it takes to obtain practice recipes and crafting books and things, but the experience gain itself is trivially quick. This causes a large number of knock-on effects:

- Rapid level-ups mean that it is hard to reward players for missions. Any reasonable physical reward is something they can probably obtain and craft themselves. If it is hard to reach high levels of tailoring, for example, then an NPC who can repair your gear becomes valuable and can be used as a mission target.
- Rapid level-ups mean that NPC followers are superfluous. However, if it takes a long time to learn a major skill, then NPC followers can fill in your skill gaps and form the backbone of a good team.
- Combat skill rising to exponential heights contributes heavily to stagnation of the mid-late game as there's little left to do to get better. In turn this leads to monster evolutions getting kind of ridiculous, quite fast, and soon in order to challenge players, we're making 30-meter tall metal zombies wielding skyscrapers.
- Players continue to operate under the not-intended attitude that their starting avatar is *supposed* to be able to learn to master everything in the cataclysm. Really, we want a game where your starting avatar feels like a normal, limited person who has to rely on allies to be able to do everything they want to do. We still want it to be possible to do that stuff... but no one person should be a master tailor, master blacksmith, master mechanic, master builder, and phd level organic chemist. Especially not in two weeks.
- The meta actively rewards players who start as strange weirdos who have virtually no starting knowledge except in specific rare, weird areas that are helpful in-game but don't represent what most people would know. It's easy to get skill, so why put points into anything besides what's hard to find books for?

### Solution you would like.

Skill gain needs to be slowed down *enormously*. The correct set point is up for debate but in my opinion:
- level 1 skill from 0 should take a day of dedicated work, or a few days of scattered reading and practice. Note that yes, this means if you start as someone who is unable to boil an egg, it will take you a long time to figure out those basics. That's because starting with 0 skill in something as fundamental as cooking is similar to starting the game with the Illiterate flaw. If you don't want this flaw, the solution is to not start out with a serious handicap. This should be relayed to the player during character creation. (note: the time here would vary depending on stuff like focus. However I'm picturing about 8-12 hours if you grind. If you spread it over a few days the actual time invested should be less due to changes recommended below)
- level 2 and 3 should take similar to, or a little less than getting level 1 in the first place. Once you hit the fundamentals, getting started is a bit easier. An average player who doesn't grind it out should reach level 3 after 2-3 weeks of gameplay. Someone who decides it's absolutely necessary to get there should hit level 3 after a few days.
- level 4, 5, and 6 should become much harder, long enough that grinding becomes a suboptimal strategy. How long this is is up for debate. I think it should be a few in-game weeks between level up, but that should be weeks where you're using the skill regularly and not grinding.
- Beyond level 6 should be ranks you don't really expect to reach in the average playthrough unless you started there. It should still be possible, but getting from level 0 to level 8 should be something reserved for very long term games.

Note that as best I can tell, proficiencies seem to be OK for the moment. These should represent a way to get lateral advancement so that there are still ways to 'level up' when your high level skill won't improve.

#### Steps to reach this point
1. Adjust starting professions to encourage players to begin at higher levels. This should be an entire issue of its own, and some PRs have already moved us this way, but in general it should be fine for a normal starting build to have one or two fairly high skills in the 4-6 range and a wide range of skills at lower levels. What I'd like to see ideally is the use of avatar age as a 'currency' for this, so the higher skills you choose at start, the older your starting character must be. I'd also like to see some of the higher level or more rare professions gated behind interesting or fun meta advancement, so if you want to be a master tailor you need to have done something in game before it unlocks as a starting option (note: the achievement in question should absolutely not be "reach level x in skill". that's boring and grindy).
2. Sanity check that there are some (perhaps default-on) backgrounds for basic adult skills everyone should have, such as "driver's license" giving 1 level of driving, and "simple home cooking" giving one level of cooking.
3. Adjust recruitable NPC skills. Diversify our starting dynamic NPC stat blocks so that there are many more of them and they have more specific skills. Rather than meeting a "cowboy" you'd meet a "plumber" with appropriate skills and proficiencies. This is a big one and also should be a pretty fun one. it could go along with giving NPCs some unique character and dialogue.
4. Ensure that skill descriptions make it clear that level 0 is very low, eg in character creation, as these are currently framed as "average". Really 1-3 is an average/beginner knowledge, 4-6 is a skilled level, and 7-10 are experts approaching mastery.
5. Fix focus. #53372 is still in my opinion a relevant and not that hard to implement solution that changes focus to reward smaller bursts of learning followed by doing other things. This should help a bit to reduce the desire to grind. In this model, whenever you practice something you get "pooled" xp that is distributed over the course of the next few hours, encouraging you to do some practicing and then go out and do something else.

Once these are done, we should be at a stage where it is safe to review the rate of skill gain. Fixing focus likely will already dramatically affect it, so we may wish to stop at this point and assess the situation before making further adjustments.

6. Add more static NPCs with useful skills and associated quests. We should guarantee that certain fundamentals in game can be found, such as a tailor who can help repair your gear, and we should tie these to both trade and story arcs. This will make it so that grinding to tailor 10 is less of a necessity and more of a playstyle choice, one that can be balanced against the desire to have other high level skills.

#### Dealing with salt
Carried to its conclusion, this will cause a lot of frustration. Any big change to the meta does. That, plus goal planning, is part of why I am putting it up here. There is no way to eliminate this; for some people, levelling up fast is what the game is about. Unfortunately it was never supposed to take a couple minutes to gain a bunch of levels, this is bugged behaviour and it does have to be fixed. However, I'm hoping that by flagging it early and addressing the things that should be put in place while we change this, it will help people adjust in advance of upcoming huge shifts.

If this change is concerning to you, consider looking in to ways to contribute some of the things that will mitigate it. Adding recruitable or quest-giving NPCs that can fix gaps in your avatar's skill base is a great way, for example, to help. Ultimately I hope that we can replace the skill-gain-grind with instead a game where the early play portion is a matter of tracking down and finding allies who can fill the gaps your avatar has in their knowledge base.

### Describe alternatives you have considered.

The exact set point of skill gains is very fluid and debatable at this point, which is why I've left it vague.

It may be a good idea to have NPC trainers able to specifically accelerate the 0-1 training speed, but I'm not certain about that.

My own thoughts on how to change focus aren't the only way, there are other options out there and some are simpler.

Ultimately though, there's no alternative to the basic fact that skill gain needs to be a lot slower.

### Additional context

All this has been a very long winded way to say "I hate fun and I hate the players".

**Comment by Alm999** [comment]
-- It would be great *if* NPCs were any good. I personally do not know a single RPG where NPC can do the talking in place of PC, say if your character has only 6 in "social" while your follower has 9. Moreover, as great as NPC may be, his/her "swimming" (AKA "athletics") is still his/hers, not your character's and NPCs do not need said skill (they don't use stamina), while PC does.

-- Also, unless I'm wrong, only PCs skill determine whether you can hotwire a car, remove/install equipment and so on.

-- As far as I know, there is no way to tell your NPC follower to pick a lock or disarm a bear trap.

Long story short: current NPC implementation is quite rudimentary and effectively stripping PC away from the ability to attain high level of skill (and setting outright prohibitive time requirements is denial in all but name) will render the game quite unplayable (unless people just debug in all the needed skills).

I must confess, I don't get the idea of enforcing even more grind as a means to "prolong game's fun". Other end-game activities can be introduced to set end-game goals, like the ability to construct "environmental purifiers" (structures that purge ecosystem from extra-dimensional invaders), organizing your own society (that would require seeking NPCs with certain backstories/proficiencies, like "Teacher", "Engineer", "Medical Doctor", "Economist" and so on): a PC can know a lot of things, but (s)he can't be everyone and everywhere at the same time… I think that would be more immersive incentive to rely on other NPCs than artificial numbers in character's sheet.

**Comment by Alm999** [comment]
What you described is basically a "party-based RPG" with different characters covering each other's needs. I'm fine with that personally, but… C:DDA has started out as a single PC "roguelike" experience. Party management, NPC AI routines even NPC controlling… Lots of things need to be changed. It looks like an attempt to  re-forge the game into entirely different beast.

To put things into perspective: currently even bringing NPCs into vaguely dangerous location is most certainly a death sentence. NPCs constantly charge into attack against superior force or flee for whatever reason, yell and complain in situations where sound is deadly (slime pits), and generally lack judgement. Ability to bring "nerdy hacker" with you into a lab or a squad of ninjas to stealth-loot a military base would require *precise* **direct control** of each and every action of NPC followers.

Ooh… making "Wasteland" out of "C:DDA" is not an easy task…

**Comment by anoobindisguise** [comment]
Should certain skills "cross train"? Like if you're a talented rifleman but never picked up a handgun or shotgun, it's easier than usual to learn those skills. Or melee making it easier to pick up bashing, stabbing, cutting and vice versa.

**Comment by I-am-Erk** [comment]
> Should certain skills "cross train"? Like if you're a talented rifleman but never picked up a handgun or shotgun, it's easier than usual to learn those skills. Or melee making it easier to pick up bashing, stabbing, cutting and vice versa.

We already have Marksmanship to represent that, and also the proficiency system for non-combat skills. I'll leave the fine tuning part to people who know more about marksmanship though.

**Comment by I-am-Erk** [comment]
> To put things into perspective: currently even bringing NPCs into vaguely dangerous location is most certainly a death sentence.

While our NPCs are definitely dumb as bricks, I've never really found this myself. However, as always, this isn't a reason to not use NPCs more, it's a reason to identify what's killing them and get them to stop doing it. Individual NPC behaviour fixes aren't beginner level code, but they're not impossibly difficult either.

**Comment by Skerald** [comment]
I see you covered your bases with the salt comment, but unless a lot changes this would significantly reduce the fun for me.

The fantasy of a slightly unhinged, unfettered, inhumanly motivated and fearless survivor reaching heights not possible in the current world and surpassing similarly big challenges is a big part of the appeal of cdda for me. Transhumanism like with all the mutagens.

Combat skill levels also are not very impactful at the moment, since the most common enemies do not dodge and later damage increases rarely reduce the hits required to kill an opponent. Everything truly dangerous gets shot anyway and most dangerous things are huge and easy to shoot. Dodge skill could be considered detrimental currently.

Also with the speed of grinding skills substantially reduced, it might reach a point where doing it is simply not worth it. What would a player do then? Scour the map for npcs meanwhile accumulating an ever increasing pile of unusable garbage or since you can not modify or repair a vehicle, leave all that loot where you found it, keeping a mental map of the places you will have to backtrack to?

Finding a good npc with the skills you need, would probably just elicit a feeling of "I can finally play the game!", instead of genuine joy. This would not increase the difficulty of the late game, but only heighten the tedium of getting there.

I suggest implementing this as a value you can change during world creation, so fun fiends like me can change it to something fantastic and people desiring a slower, perhaps more realistic experience can enjoy it as well.

**Comment by TheShadowFerret** [comment]
>The meta actively rewards players who start as strange weirdos who have virtually no starting knowledge except in specific rare, weird areas that are helpful in-game but don't represent what most people would know. **It's easy to get skill, so why put points into anything besides what's hard to find books for**?

I thought the plan for this was "yeet the points some time after #64697 was merged for being arbitrary"

**Comment by I-am-Erk** [comment]
> The fantasy of a slightly unhinged, unfettered, inhumanly motivated and fearless survivor reaching heights not possible in the current world and surpassing similarly big challenges is a big part of the appeal of cdda for me

I feel for you, but that's always been incompatible with the design goal. We're not building a post apocalyptic power fantasy. The goal is to make it feel like you're a person actually surviving in an unbelievably weird apocalypse.

**Comment by Fris0uman** [comment]
> > The fantasy of a slightly unhinged, unfettered, inhumanly motivated and fearless survivor reaching heights not possible in the current world and surpassing similarly big challenges is a big part of the appeal of cdda for me
> 
> I feel for you, but that's always been incompatible with the design goal. We're not building a post apocalyptic power fantasy. The goal is to make it feel like you're a person actually surviving in an unbelievably weird apocalypse.

There will probably still be room to add trait/artefact/bionics to be an unnaturally good learner I guess, and if not in vanilla maybe in a mod. This looks like a good ground for a perk for exemple. Making learning fast a special reward instead of the norm seems extra interesting to me.

**Comment by PatrikLundell** [comment]
I'm not opposed to most of the suggestion as such, but things have to be done in a reasonable order, i.e. make sure replacement strategies are available at the time skill progression is nerfed, rather than leading with nerfing skill progression and then slowly and gradually add the means to deal with it later.

Thus:
- Lead with the introduction of useful recruitable/employable NPCs. They cause no harm in the current game, even though many may find them to be of little value when they still can do everything themselves.
- Follow up by making NPCs useful in dangerous situations. If you strip away the ability of the PC to deal with danger by nerfing their ability to become skilled and get good equipment, compensate by making it possible for the group to deal with enemies. This is difficult, so it might be reasonable to nerf crafting at a different (earlier) time than activity oriented skills, which may lead to two parallel progression systems for a significant amount of time until unification may be reasonable.
- As far as I understand PC character swapping is currently not a main game option. It probably will need to be made a normal main game functionality, especially during the (probably rather lengthy) period during which the ability to order companions to do tasks for you remains absent or only partially implemented (picking locks, hacking terminals, specific vehicle repair to make them usable rather than repair everything which you don't have resources for, etc.). It can also be noted that character swapping will be a major game play focus change some people are likely to be rather unhappy with (I prefer party based games, so it would be a benefit for me).
- Nerfing grinding has to be done with care: You can't really be expected to enter a lab and do a little bit before deciding that this is enough training, and in order to get anywhere skill wise, the grinding has to be done by snipping action activities into little parts and grind those by doing a little bit, leave, and then come back a week later to grind the next snipped. In general, when you expend the effort to travel to a place, you should not be penalized for dealing with a location in one session (not saying that you should never have to retreat due to injury, lack of ammo, damaged equipment, etc., only that you shouldn't be penalized for not cutting it up).
- I remain adamantly opposed to locking things up behind grinding solely to allow you to start from where you want to start the actual game play and its grinding. That goes for starting scenarios as well as for starting skills or professions.
- If player combat capabilities are nerfed monsters should be nerfed at the same time, not as something to be corrected later by somebody else.
- Also note that nerfing has to be compensated for with simultaneous reduction in the base monster evolution rate. If the progression rate of the player is reduced, so should the progression rate of the enemies be reduced. As long as this rate is a parameter, experienced players may adjust it (but only after wasting time and effort on trial and error to determine what a reasonable rate is), but inexperienced players may be screwed over when their gear and skills barely have changed when the hulks appear in numbers.
- I would suggest some skills should be considered standard, so characters would default with a level of 1 in them. I would expect almost everyone to have a basic understanding of cooking, sewing, mechanics, and athletics, for instance, unless they somehow are males from a "traditional" gender separated society for the first two, and females for the last one, and such people ought to be rather rare in New England. A level of 0 in those skills ought to be challenge modifiers.

@Alm999: Mask of the Betrayer had a system where companions could interject in dialog when their skills were superior to that of the PC, and some other games have had companions take over the talking (the demo of Broken Roads allowed you to hand the talking part over to the negotiator companions, for instance). So there are examples, but in this game I would suggest character swapping would be the way to handle that, although you probably wouldn't be able to swap in the middle of a conversation (which affects how skill checks could be used in dialog, at least until the skill check logic is expanded to allow it to check for skill levels of available companions).

**Comment by estebandellasilva** [comment]
Point 1 : Then skill decay should also be slowed down massivly
Point 2 : maybe its possible to assign certain proficencies to a skill ... 
at lvl 3 you should have 3 base proficencies related to that skill
at lvl 5 you should have 3 advanced and 6 base
at lvl 7 you should have 3 mastered and 6 advanced and 9 base.
for lvl 10 you should have mastered at least 6 proficencies related to that skill

another way it could be handled iss like without proficencies you can reach lvl 5 as a hardcap - certain proficencies which are ... related to that skill could increase the hardcap -> until you can reach the maximum

**Comment by Alm999** [comment]
@estebandellasilva I believe the idea is to make Lv.10 virtually unobtainable so if you really need a high level of certain skill (and haven't chosen that skill as your starting profession) then you shall seek a skilled NPC.

Take a "computers" for example. For that you must rescue a dedicated hacker who was meddling with computers since age 8. At 16 she tried to hack the Pentagon, got arrested and sent to prison, where she managed to hack local security system and organized a breakout. Later she worked for crime bosses stealing money from stock exchanges and in the process robbed all 6 of her "employers" but was discovered and had to run, forging her new fake identity, was finally caught by FBI and escorted to prison (again) but managed to escape by utilizing a backdoor in police bot OS placed by her back when she still was trying to hack Pentagon but managed to get into Uncanny's R&D department's server. Oh, she is at age 37 and rated at "computers 7", BTW. :)

**Comment by PatrikLundell** [comment]
It can be noted that you tend to need to gain quite a few proficiencies to gain levels in the current system, as the recipes available to train tend to require proficiencies, so you have to grind their prerequisite proficiencies to get a reasonable chance of producing the items (which you may or may not have use for). I think this kind of natural dependency gating is better than an artificial one based on number of acquired proficiencies (and it may be difficult to invent enough proficiencies for some skills, such as athletics. Top performing athletes tend to be very narrow in what they're experts at).

I do agree skill decay has to be eliminated or decreased vastly if the system itself makes it inefficient to gain skills by training them daily, or you'd spend most of your skill using time regaining lost ground.

A problem with a small pool of dedicated recruitable skill providing NPCs is that the game currently is very weak when it comes to protecting them from getting killed or end up in inaccessible locations (because there are no routes to their locations, or the routes are blocked). Basically the same issues that plague unique locations, but with even less protection.

**Comment by estebandellasilva** [comment]
> @estebandellasilva I believe the idea is to make Lv.10 virtually unobtainable so if you really need a high level of certain skill (and haven't chosen that skill as your starting profession) then you shall seek a skilled NPC.
> 
> Take a "computers" for example. For that you must rescue a dedicated hacker who was meddling with computers since age 8. At 16 she tried to hack the Pentagon, got arrested and sent to prison, where she managed to hack local security system and organized a breakout. Later she worked for crime bosses stealing money from stock exchanges and in the process robbed all 6 of her "employers" but was discovered and had to run, forging her new fake identity, was finally caught by FBI and escorted to prison (again) but managed to escape by utilizing a backdoor in police bot OS placed by her back when she still was trying to hack Pentagon but managed to get into Uncanny's R&D department's server. Oh, she is at age 37 and rated at "computers 7", BTW. :)

The Problem iss people dont like skill decay ... and reaching lvl 10 iss the only way to stop skill decay ...
Another thing is what many  people pointed out ... you cant rely on npcs because they arent that good -> until you make NPCs not a liability but forcing the other change you would do basically what was done with portal storm again

**Comment by esotericist** [comment]
skill rust is only the rate it is because skill gain is at the rate it is. it seems fairly obvious that both would need to change. rust is almost certainly not going away, though, not unless some other mechanic that fills similar purposes gets implemented. (which seems pretty unlikely)

**Comment by Alm999** [comment]
Skill rust is completely tangential to this. As I understand, the idea is to make PC unable to be master at everything. Player shall choose starting skills/proficiencies (and they shall be adequately developed: no police officers with only 2 in "handguns") and all other (not pre-selected) skills shall be immensely hard to grind. Like, if your PC is not a career athlete, it shall take several (probably, in-game) years to train "athletics" to 5; your character needs to dedicate whole life to a field of expertise to gain significant progress.

So, if a need for a skill arises, player shall seek a competent NPC that would cover PC's needs in the required field (medic, hacker, burglar, sapper, mechanic and so on).

**Comment by I-am-Erk** [comment]
We aren't really looking at *years* to train up a skill. A person can become pretty competent (level 4-5) in most things after a few weeks to months.

**Comment by Alm999** [comment]
The exact time frame for learning a skill up to a mediocre level (and lv.5 is mediocre in the most literal way possible -- it is in the middle between the worst and the best) is not that important. Required time can be adjusted.

What I described is the idea as I got it: high levels of skills that do not stem from the pre-selected PC's hobbies/occupation shall be nearly unobtainable (or at least made hard and tedious to train) to incentivize players to seek NPCs' help. And to that end NPCs are planned for a general rework to make them actually useful and reliable in all situations. Well, as reliable as reasonable, of course. They shouldn't be un-killable and unstoppable, but all their deaths shall be the results of player's wrong decisions or actions, not AI stupidity.

**Comment by keampe** [comment]
I absolutely agree with greatly slowing down the skill progression as right now it's ludicrously fast. 

One thing to consider, though, is character creation and what is considered average. If you open up the game and create a new custom character you will see that Offense, Defense and Social are all average. You haven't even selected attributes, traits or skills. So, 0 skill is currently considered average. for these. For Knowledge 4-5 total skill levels amongst all crafting and interaction skills is considered average. This is before attributes and traits are added in. This is in no way representative of average people.

So, a new player is going to look at this and think that you are supposed to start the game with a handful of very low skills, which actually represents anything but an average person.

Average people should really have 2-3 points in every skill that they use occasionally (or all the time but not for particularly hard tasks like driving to work) and at least 4-5 in skills they use to make a living, probably higher. Take Police, for example - the Police interceptor has a Vehicles of 3 - same as the bus driver - and none of the others has any better than that, with most police having none. All that despite police being trained in pursuit and advanced driving as a matter of course.

tldr - I think a revamp of the way skills are perceived would help players get used to starting with higher, more realistic, levels of skill. This would in turn lessen the hit of slower skill gain and would reduce grind in general, especially among new players.

Just some thoughts.
   - Shane

**Comment by Skerald** [comment]
Just had a thought on melee combat skills:

You should not be able to get high melee skills from character creation, but improvement should be somewhat faster. Why?

Post cataclysm melee skill ≠ today's melee skill.

If you train in some martial art today, you are quite limited in your realistic training opportunities and rarely practice one, that is intended to kill. Even more informal combat is usually not meant to kill someone and even if your character made a living of killing people with his bare hands or a melee weapon, their opponents were likely lacking and fights ended quickly.

Also martial arts are designed to fight humans, not wasps or zombies. Self defense tactics will most likely fail against zombies, who feel no pain and are immune to exhaustion. You may be proficient in subduing an unarmed survivor, but how often does that happen? 

As for weapon skills: Let us say you did train HEMA and know how to use a sword. Would you know how to extract your weapon quickly, once it got stuck in a ribcage or how to prevent that? How much force is enough to kill and what would just tire you out? Perhaps fully impaling a zombie through the chest just gives it the opportunity to chew your face off? You might know how to defend against swords and spears, but unless you start in a mansion, that knowledge is going to be mostly useless.

The very basics like distance management would of course still be valuable, but the things that would make up a high melee level for a character just can not be learned today.

Surviving a week and killing dozens of zombies will likely trump any experience or training a character has received previously.

**Comment by Dhanvanthri** [comment]
I think sleep/rest/downtime should play a bigger part in the remade system as well. I'm sure I'm not alone in the feeling that studies, and practice in general only solidifies over time and with good rest. I think this is true basically across the range of things I've done, from comp sci, mathematics and philosophy to guitar, skateboarding and shooting.

In addition to the thoretical and practical difference, you should gain a portion of the experience frontloaded and some of it over the course of time that you sleep, and generally not focused on a task. This passive over time component could be more random/jumpy. 

In certain activities, like fighting games, I come back after many weeks much better than I was before just from the break in my entrenched patterns coupled with a fresh perspective.

**Comment by I-am-Erk** [comment]
> I think sleep/rest/downtime should play a bigger part in the remade system as well. I'm sure I'm not alone in the feeling that studies, and practice in general only solidifies over time and with good rest. I think this is true basically across the range of things I've done, from comp sci, mathematics and philosophy to guitar, skateboarding and shooting.
> 
> In addition to the thoretical and practical difference, you should gain a portion of the experience frontloaded and some of it over the course of time that you sleep, and generally not focused on a task. This passive over time component could be more random/jumpy.
> 
> In certain activities, like fighting games, I come back after many weeks much better than I was before just from the break in my entrenched patterns coupled with a fresh perspective.

That is basically the goal of the focus rework I described, for the short term part, and the reason for the xp boost given by recovering from skill rust.

**Comment by Dhanvanthri** [comment]
> 

From reading that issue, you're clearly some parallel universes ahead of me XD
Cheers m8

**Comment by Quacknis** [comment]
Quoting someone i saw 

>If you give the average person the recipe for a cake, and the tools and ingredients required to complete it, they shouldn’t have to attend culinary school or read 8 cookbooks just to give it a try. In regards to crafting / construction / mechanics, you should be able to try crafting anything regardless of skill level, so long as the recipe, tools, and ingredients are available. Skill levels should decide the success rate, completion time, the effects of crafting failure (minor / major mistakes), and whether or not you actually need a recipe.

I completely agree with this take, capping skills to try things doesn't make any sense. The skills should only dictate failure and time needed. Though i guess this is overlapping with profficiencies. Still  why would i need to boil 20 eggs to try fried liver? The whole recipe system and the needed experience for reach recipe would have to get a huge look over as well.

**Comment by Dhanvanthri** [comment]
> > I think sleep/rest/downtime should play a bigger part in the remade system as well. I'm sure I'm not alone in the feeling that studies, and practice in general only solidifies over time and with good rest. I think this is true basically across the range of things I've done, from comp sci, mathematics and philosophy to guitar, skateboarding and shooting.
> > In addition to the thoretical and practical difference, you should gain a portion of the experience frontloaded and some of it over the course of time that you sleep, and generally not focused on a task. This passive over time component could be more random/jumpy.
> > In certain activities, like fighting games, I come back after many weeks much better than I was before just from the break in my entrenched patterns coupled with a fresh perspective.
> 
> That is basically the goal of the focus rework I described, for the short term part, and the reason for the xp boost given by recovering from skill rust.

From reading that issue, you're clearly some parallel universes ahead of me XD. I neglected to read that link in the OP
Cheers m8

**Comment by Tharn** [comment]
> > The fantasy of a slightly unhinged, unfettered, inhumanly motivated and fearless survivor reaching heights not possible in the current world and surpassing similarly big challenges is a big part of the appeal of cdda for me
> 
> I feel for you, but that's always been incompatible with the design goal. We're not building a post apocalyptic power fantasy. The goal is to make it feel like you're a person actually surviving in an unbelievably weird apocalypse.

For the first X years of the game's development, that is what the game was. We were given the tools to get from a modest start all the way up "to a 10", to use the skill ceiling analogy, and now we're meant to hit that ceiling at 6. The game is pivoting to be something else and you don't expect the people who enjoy what the game used to be to be mad about it, or right to be mad? That does not compute for me.

Skill changes to aid realism are fair enough, but consider that the development lately has been a chain of many "realism nerfs" to various systems and these *compound* on each other. It's getting difficult to see the *fun* in there amidst all the realistic chore-type stuff.

What's so terrible about a long survival career awarding the player with excelling at a certain skill? You can make it hard; you could even introduce systems for PCs to not be able to max every skill, and you could have NPCs that are useful in their specialty fields without taking a hacksaw to player character potential.

Lastly, players will keep grinding and slowing down skill progression is not the answer. They will just have a worse time doing it. Frankly, I don't see too much wrong with the way that skills progress in 0.G.

I keep reading "NPCs should matter more" in various wordings, and can't help but feel that the reasons NPCs don't matter currently is that NPCs are still barely working. It's got little to do with how much of a skillbeast the PC is or isn't, but that, you said it yourself, NPC AI is basically not there. Camp stuff is in alpha stage. Who plays with NPC needs enabled? The player has to "game the game" in weird ways to keep NPCs alive; they spawn in overrun cities where they should not survive and often die the minute they enter the player's reality bubble. They're either suicidal or timid. The erratic introduction part where you're not sure how it's gonna go - that part is great fun at least. If you're looking to invest time and energy into big ground-work type changes, please consider NPCs. That will add some of the value you're looking for.

> All this has been a very long winded way to say "I hate fun and I hate the players".

True, except there's nothing funny about it. Taken at face value, this issue should be closed outright since it doesn't seem to aim at improving the game.

**Comment by DeciusBrutus** [comment]
If making a packet of instant Ramen using an electric kettle and water is a task for Cooking level 1, then level 1 isn’t the baseline level that fully functional adults have, it’s a disadvantage that some people who have limited life skills have. 

I think a good metric for the baseline would be to look at the things that someone does regularly and make sure that their skill is high enough that they no longer gain skill from their regular tasks. That means that a plumber would inherently have more than vehicles 1, since they drive every week. 

A starting value of 8+ in one or 7+ in two skills should be pretty common for professions that are literally professionals, and if it’s odd that a plumber gains welding and electrical knowledge above their level that’s a reason to break “fabrication” into multiple skills, not a reason to require that a professional have several trades before being skilled at their primary one. 

I would like to see skill books that allowed or helped in specific tasks above the general skill level; a book that provides specific instructions on how to make explosives wouldn’t need to train applied science generally, just allow the use of specific recipes better than reading college chemistry textbooks and doing the lab work until you understand from first principles how to make explosives. However, just following written directions doesn’t provide the same learning experience. 

Having crafting, construction , and vehicle modifications use the best of all skills and all known recipes among nearby allied characters would mitigate a lot of the crafting-related issues, at a programming cost that is most of the complete overhaul those systems need. It might be worthwhile to design how those systems need to end up and make a skill system design that works with them first.

**Comment by Zireael07** [comment]
@Alm999 I think you and @I-am-Erk had a misunderstanding, because "mediocre" doesn't mean "average" or "middle of the way" as you intended, it means https://dictionary.cambridge.org/dictionary/english/mediocre

**Comment by Elshad19** [comment]
Please, check this [post](https://www.reddit.com/r/cataclysmdda/comments/15phcmx/big_changes_to_skills_are_on_the_horizon/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=1) on reddit before you start working on this issue.

**Comment by I-am-Erk** [comment]
If we're at the point of this being a Reddit salt thread, I'll close it for a bit. It's produced plenty of useful feedback that had altered the initial post, but that's pretty much the opposite of what a Reddit salt thread about how mean the contributors are is going to do. There's a very good reason all the Devs left Reddit and you can see it right in the third post of that junk.

If you have important observations in the vein of the one about character generation feedback, please let me know on discord or similar

**Comment by I-am-Erk** [comment]
To comments. Close to comments, sleepy brain

**Comment by I-am-Erk** [comment]
> If making a packet of instant Ramen using an electric kettle and water is a task for Cooking level 1, then level 1 isn’t the baseline level that fully functional adults have, it’s a disadvantage that some people who have limited life skills have.

I think there's room to review what's a level 1 requirement that should be level 0, but you'll note in the OP I already commented that players should start with higher skills and a wider variety of skills 

Generally a character should be starting much closer to the desired set point, not starting with almost no skill and learning everything after the apocalypse.

**Comment by ZhilkinSerg** [comment]
> Please, check this [post](https://www.reddit.com/r/cataclysmdda/comments/15phcmx/big_changes_to_skills_are_on_the_horizon/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=1) on reddit before you start working on this issue.

No

**Comment by I-am-Erk** [comment]
I'll probably reopen this tomorrow as the worst of the salt has died off in other communities. However, be aware that if your comment is just a complaint about realism or how I must hate fun to demand a change like this, I'll know you haven't read the issue and instead are just coming over here from one of the dozen-ish Reddit threads. If you have arguments against the suggestion outlined, address the suggestion outlined, not whatever memish ideas you've gotten from an online troll post. If your comment gets deleted, that's almost certainly why, and if I have to delete more than one or two comments, I'll close it again for a while.

If you've got actual feedback about what other areas I've missed that should be prioritized, I look forward to hearing from you.

**Comment by Qrox** [comment]
Before I lay out my suggestions about your proposed changes, I suggest you move your "dealing with salt" section to the start (and give it a less inflammatory title) so that people understand what you truly intend to do without having to parse through a huge wall of text. Not everyone is a developer interested in and understands game development and this is a public forum for everyone to see after all.

And before I start my own wall of text, here's a brief summary of it: While the proposed changes are very ambitious and promising in theory, it still lacks multiple huge overhauls before it can actually make the game more fun and believable instead of less (as you've admitted to in the later sections of your issue), and some of the proposed changes might actually make people grind more instead of less. Given the community nature of the game development, I'm not holding my breath before the game is in a state that the proposed changes make sense, and I hope we do not rush any of the proposed changes only to make the game more broken (which I see you're planning to avoid, and I appreciate it).

And here's a point-to-point review of your proposed changes:

> For example, it takes 3 minutes in game to advance to skill level 1 from untrained ...

The skill gain speed at low levels is definitely too high and needs adjusting. (I was able to level shotgun from 0 to 2 in one practice session so that might be an good start for a fix).

> Rapid level-ups mean that it is hard to reward players for missions ...

That sounds like a fun system, but it would require first making NPCs to be able to craft/repair/do whatever things for you (as you've proposed in your recommendations), and I'm not sure that will be any time soon given that these existing systems are all implemented as mostly independent systems. Before that happens, restricting the player's own ability to do things will only make certain things purely impossible or slow them to a grind if they do not choose a high enough starting skill.

Speaking about awarding players for missions, currently NPC skill training seems to be taking a ludicrously long time and there's no indication of the training progress nor a way to interrupt the training and be able to resumed later.

> Rapid level-ups mean that NPC followers are superfluous ...

Agreed.

> Combat skill rising to exponential heights contributes heavily to stagnation of the mid-late game as there's little left to do to get better ...

While I do agree, I don't think that's the main issue of the current state of the game. The thing is that the games systems are so unbalanced that you can basically ram hulks and grind them into a fine paste with your death mobile while diving into a lab makes you get beaten to death by one zombie predator. Fixing the combat skill learning rate won't help fix the first issue and will only exacerbate the second one.

> Players continue to operate under the not-intended attitude that their starting avatar is supposed to be able to learn to master everything in the cataclysm ...

Well, the old system could be hand-waved as everyone's intelligence being enhanced by the blob just like how the healing system makes you heal battered arms overnight, so I won't blame the players for not liking the system change.

And my argument is that people play a game to experience what the game has to offer, not to relive their years persuing a degree, so a game should allow people to master everything possible in game in a reasonable amount of time before it becomes a grind, especially considering there isn't actually that much real playable content behind some of the game's skills.

> The meta actively rewards players who start as strange weirdos who have virtually no starting knowledge except in specific rare, weird areas that are helpful in-game but don't represent what most people would know ...

While that's true, I'm not sure if the currently proposed changes are going to change that. People who were trying to start with the most useful and hard-to-obtain skills will continue to do so, and I double anyone has the feat to balance the game to the point that every skill has the exact same utility-to-price ratio.

> Skill gain needs to be slowed down enormously. The correct set point is up for debate but in my opinion ...

One problem is that currently one level represents different amount of knowledge in different skills, so either someone rebalances every skill to a strictly defined level scale (which will take a huge amount of effort and won't be done any time soon), or they calculate how much utility a given skill level has for every skill and adjust the learning time accordingly for that level (which should be less difficult but still requires considerable efforts). Before that any attempt at changing the learning time between levels is only likely going to fix one skill and break every other.

Also, while this might encourage some people who grind to give up on grinding, this will also make people who do not grind unable to level up their skills simply by doing what they have to do, which I'm afraid might actually make people grind more instead of less, unless someone devises a perfect system where people actually have better things to do before leveling their skills up by grinding.

> Adjust starting professions to encourage players to begin at higher levels ...

It's unclear to me how this differs from the current system. First, I think people can already begin at higher levels. Second, it does not solve the problem you mentioned about people choosing weird starting skills if we allow player to choose their profession freely.

> Adjust recruitable NPC skills ...

> along with the above, we need to ensure NPC's in party are actually assisting with tasks they should help with ...

> Ensure that skill descriptions make it clear that level 0 is very low ...

Agreed.

> Fix focus.

I haven't read that issue so I don't have any comments for now.

> Add more static NPCs with useful skills and associated quests ...

Agreed. These will require huge overhauls of the NPC system though, as I've said earlier in this comment.

> Break out the old nerf bat and earn your keep in the salt mines.

I hope this will truly be the last step as advertised because based on previous development history I'm not totaly conviced this will be the case.

**Comment by I-am-Erk** [comment]
Qrox: note that each individual step, including busting out the ol' nerf bat if necessary, may take a while and go through many steps. This is an approximate outline.

I disagree that static NPC crafting additions need all that much special work though. We already have everything necessary to meet a new NPC, earn their trust through missions or dialogue, help them obtain materials for crafting special equipment, pay them for the equipment, and return when it's done. It's a bit tougher with something like mechanics or gear repair I suppose.

I'd highly recommend checking out the focus issue, as it's pretty crux to the whole thing and I'd really like some more eyes on it. For example it addresses, or intends to address, a fair bit of what you're concerned about with people who don't grind potentially being unable to gain skills, in that they'll gain a fair chunk more XP overall for their actions spread over a longer time.

**Comment by estebandellasilva** [comment]
2 Things that maybe should be taken into consideration

1: For Crafting when proficencies are available ... if you reach certain proficencies they should help with skill gain, if you gained for example 6 out 10 available proficencies for Farbrication you cannot say that the person iss not somewhat knowledgeable about fabrication - so skill level 6 should not be the the "softcap" he can reach (if he started with zero skill)
2: To be able to reach level 10 in any combat skill currently requires you to specifically grind certain enemies are weaker enemies do not give you any experience -> if you just increase the time it takes without taking into account with which monsters can be used to train level 10 will be absolutely unachiveable (even for guys who start with higher skills)

**Comment by cm97878** [comment]
When I read some of the various threads on the drama of the week I got an inkling that there was some catastrophising going on, and having read through this all here I'm glad to see I was right. Not to say I personally agree with all the changes, I'm definitely firmly in camp "likes to be good at everything" - but, on one hand with how this was described it sounds like (with the proper NPC changes and infrastructure updates) I can definitely see myself loving it, and on the other hand if it doesn't vibe with me it definitely wouldn't be too hard to mod skill gain up and pretty much fix the issue I'd have.

I am curious though, what does the dev team see as being the general progression with this, from a purely early-planning-stages theoretical point? Specifically in terms of crafting skills, as practical skills like lockpicking or trap-disarming or what have you seems pretty cut and dry - bring em with you and either order them to do X or swap control and do X.

But as for crafting, I can't imagine the goal is to just swap to an npc to craft for 8 hours as your PC sits there with a gormless stare. First answer that came to mind (for someone basically entirely unfamiliar with the code) would be something like the bill system in DF/Rimworld - workstation/designated crafting area, context menu to queue up crafts to be done. But, I'm not sure if this is remotely possible with the current way things are structured. Are there any tentative plans for how to handle longer-length tasks like this w/ npcs?

**Comment by I-am-Erk** [comment]
The first and easiest crafting change I'd think of would be to hire a static NPC to do it. So like, we make a "Zug the tailored" you can meet and pay to do commission work. However there's also already a mechanic in to get your followers to craft for you that can be developed further.

Esteban: good point about making sure combat isn't a vile mess with this. Better tools for sparring and weapons training probably would fit there

**Comment by PatrikLundell** [comment]
To expand on I-am-Erk's answer for those who aren't up to date: There was a recent addition that made it possible to directly order a companion to craft items they have the skills to craft, while your character does something else. It has worked very well for me so far, although I haven't tried to see what happens when you move so they end up outside of the reality bubble, nor have I given crafting tasks that take a very long time (that should result in a loss of visibility during the night and need to sleep). Note that this is distinct from the base camp companion crafting system which remains in place (but certainly could use a total rework).

**Comment by I-am-Erk** [comment]
Thanks for that, I'm on trains and things today and didn't have time for a longer reply.

**Comment by Termfin** [comment]
Hey Erk, I understand that this change is very much still in the drawing board phase, but I was wondering if you had a general idea of how NPC spawns would work with the rebalanced skill system? 

Currently, by default, NPCs spawn fairly rarely; save for overmap specials like the refugee center. Depending on world generation, this makes the game's atmosphere rather lonely. What is the vision for how often a player should encounter someone else? And how often should a player expect to meet an extremely skilled NPC?

Thanks.

**Comment by PatrikLundell** [comment]
Note that the NPC spawn rate is a parameter the player can set (I generally reduce the spawn rate significantly because I find there are too many of the currently usually rather useless NPCs to deal with). At something like 12 days into my current game, with a reduced NPC spawn rate. I've already encountered (seen on the map, not necessarily approached) about half a dozen without even bothering to visit the refugee center yet. I suspect some NPC spawns don't respect the spawn rate parameter, but rather have a hard coded rate (I've got two or 3 survivors in evacuation shelters, for instance). Note that there are bandits on top of that.

My understanding is that the random NPCs encountered should be adjusted to have rather normal sets of skills with a modest bump in the skills related to their professions. Extremely good NPCs should probably be very rare among randomly generated ones, and instead you'd find out about them from other NPCs or stumble upon their custom generation locations. Some of them probably wouldn't be recruitable, but rather be available for performing tasks (akin to the the two artisans). Obviously my take on it, so it's not in any way any official stance (I have no official capacity).

I wouldn't be opposed to a way to select the profession of the starting location NPC (for scenarios that start with an NPC, obviously), especially initially, since I'd expect most players to want to have a fighting oriented PC, which means companions (if you manage to recruit them) would be most useful if they brought other skills to the table. Once companions become more useful and reliable (as in being reasonably sure they behave as instructed or in a reasonable and somewhat predictable way) the PC may change to having some other priority, but that's probably a fairly long way off.

**Comment by DeciusBrutus** [comment]
I think that NPCs should have a starting profession taken from the starting professions available to the player, possibly excluding some of the more fanciful ones or possibly restricted based on location, and then have equipment, skill, and possibly mutations and such altered by time since the cataclysm. 

If a plumber PC starts with 7 fabrication, a plumber NPC shouldn’t start with less. 

I also think that NPCs that the player finds after a long time should have done some things to their surrounding area, like pulp zombies and fortify, but that feels like it involves dealing with mapgen in a complex way.

**Comment by Fajdek** [comment]
How will this change affect books and theoretical xp gain?
When I first started out and read fabrication 0-2 books, I found that it taking a few days makes sense, but then I found carving practice and it lets me hit fabrication 2 pretty much instantly, which is what this issue wants to solve.
On that note, will book reading times be changed, either slightly or massively? The skill gain from crafting stuff, such as sundress really needed a nerf as I could obtain tailoring 3 in a matter of a few game hours from 0, whereas reading the same tailoring book 0-3 would take me atleast 5 days in game, so I don't think books really need a huge nerf.
Keyword: Huge, this issue wants to make it apparent that everyone should have some basic level of understanding of everything, and level 0 is your character literally not knowing anything in the concept, so I would understand if books would receive a small nerf.

**Comment by I-am-Erk** [comment]
@termfin: it would be nice to improve how dynamic NPCs spawn and wander, but for the purposes of guaranteeing certain skills are available I'd also like to add more recruitable NPCs that spawn in specific maps and can be found through quests and dialogue. Eg. Learning about other survivors that the merchants trade with, and getting to meet them.

@PatrikLundell I also would really like to see the starting npc be more customizable, I'd also like it if they could have a defined relationship to your player.

@DeciusBrutus the player starting professions presume you're going to also take some backgrounds. Rather than recoding it, I'd prefer to design a broad set of general professions that cover a few bases, so that it's easier to give them each some more interesting details and dialogue as well. See the Motorhead template I'm working on now.

@Fajdek focus changes would apply to books as well. Whether or not they'd need more changing after that, it's much too early to say.

**Comment by I-am-Erk** [comment]
Ooh, just had a cool thought. The Teamster should have a dialogue option, basically "looking for group", where you can ask merchants to send out word you're looking to recruit someone, and they'll bring back either a prospective recruit with them, or a mission to get one. Possible option to specify you're looking for a particular skill set.

**Comment by esotericist** [comment]
@Termfin side note about npc spawns: in addition to the world setting slider mentioned above, it's important to note that npc spawn chance slows as the number of npcs in the world *near you* goes up. if you travel more and longer distances, more of them spawn.

they also gravitate to 'safe' locations. some of which admittedly aren't very safe. at this time, other than the player loading them into the reality bubble (where the zombies or mutants presumably eat them), they won't die off naturally, and continue to occupy 'population slots' for lack of a better word. (there isn't actually a hard cap, but probability approaches nil after a while)

**Comment by PatrikLundell** [comment]
I like the Teamster dialog suggestion, but would have that resulting in a "quest" of the prospect's location rather than having the NPC delivered (and potentially rejected, stranded with the Teamster).

Also realized there would need to be general "potentially recruitable NPC" dialog where you can ask them about their skills (currently you can look at them to assess stats, but there's no way to get at skills without recruiting them first). I would gate such dialog behind a trust and/or PC faction reputation and/or recruitment quest check (and provide means to gain sufficient trust).

**Comment by I-am-Erk** [comment]
Discussion of the teamster quest idea should be saved either for the pr where it gets added or an issue post suggesting it. It's mostly a digression here, though it helps this concept. I'm going to try to finish off the motorhead pr and then work on that teamster quest on my enormous drive today so maybe you will see it soon.

**Comment by Raikiri** [comment]
Personally I'd rather play a gimped character who will likely die trying to survive alone rather than rely on NPC's that can just win the game for me.

**Comment by ZhilkinSerg** [comment]
> Personally I'd rather play a gimped character who will likely die trying to survive alone, rather than rely on NPC's that can just win the game for me.

Nothing prevents this kind of gameplay (except there is no winthegame)

**Comment by I-am-Erk** [comment]
Nobody's going to force you to rely on NPCs. You'll just find being a lone wolf to be an interesting challenge run rather than a default play style, because a single person won't be easily able to master every skill at a moment's notice.

**Comment by Raikiri** [comment]
Buying welding rods from the refugee center in thousands, straight up buying all the medieval weapons/armor for stuff I don't need? That's basically "winning the game" compared to the alternative of crafting all of it manually for months even with the current skill gains.

But I didn't want to sound too negative about it : I always enjoyed the difficulty of playing a lone wolf character and I hope I'll still enjoy it when it gets harder.

**Comment by I-am-Erk** [comment]
If you feel that the balance of merchants needs work, which it very probably does, feel free to explain how we should change it up as a priority before skill changes. Otherwise it's hard to read that as any thing but very passive aggressive. The whole purpose of this issue is to identify key things that need fixing for this to work, so like... Yes? There are issues that need fixing for this to work.

**Comment by Raikiri** [comment]
I think I can explain my reasoning. Imagine a part of the game that you really like. I really like raiding TCL's, so let's use it as an example. Now imagine there's a new NPC who you can pay to raid a TCL instead of you doing it yourself. Why on earth would you want to do that for any amount of money? It does not matter if it's cheap or expensive, that's what you like doing, why on earth would you pay somebody to do less of it?

I guess people just like fixing their cars, people like crafting their stuff, so they don't want to delegate the part of the game that they enjoy to somebody else (and even pay for it?).

Pardon me for sounding salty. I can't pretend to sound excited about changes that don't make me excited. But I do genuinely want the devs to pursue their vision, be passionate about it, because if they like their game in the end, I'll probably find a way to enjoy it as well.

**Comment by I-am-Erk** [comment]
You're pardoned for sounding salty. I know where you're coming from, but the whole objective here is to try to ease things in a bit. The core concept - skill gain needs to be slower - isn't negotiable, but *how we make that possible* is, and my whole point here is that we should do what we can to prepare for a big nerf like that specifically to avoid having to just drop it and watch people scrape up the pieces.

The plan I've got isn't that you *now must* pay an NPC to fix your car or repair your equipment (or loot your TCL or anything else), it's that you should be able to choose to not have to do those things and have someone else able to do them as part of a modified gameplay loop. It's not a substantially different game to design  your death mobile in the same interface, then pay an NPC to make it while you go out looting... But if you don't like it, then you can always start as a novice mechanic and train your way up, or start as a pro mechanic and not even have to do that. The key is that there will be choices. You don't *have* to grind mechanics if you want to build a deathmobile.

I'm entirely open to alternate suggestions, but npcs who can help you do the work seem to me like the most accessible and logical option

**Comment by Raikiri** [comment]
In my opinion, there's absolutely nothing wrong with straight up nerfing skill gain -- becoming a master in any field in real life takes much more time than just grinding something for a couple days in a row. However, in real life you don't need to meet any specific threshold before you can start making your chainmail. In real life you can just get your pliers, a bundle of wire and start making your first really crappy chainmail -- it's not gonna turn out great, but nothing really stops you from doing it.

In real life nothing really prevents from trying to bolt a boom crane to your vehicle. You're quite likely to injure yourself, damage your boom crane and also the vehicle, but you're quite likely to achieve it in at least some way without spending a year to attain a PhD in attaching boom cranes.

**Comment by Dhanvanthri** [comment]
I think another thing that would be nice is if skills had different rates of rust/catchup.

**Comment by DeciusBrutus** [comment]
> 
> In real life nothing really prevents you from trying to bolt a boom crane to your vehicle. You're quite likely to injure yourself, damage your boom crane and also the vehicle, but you're quite likely to achieve it in at least some way, even without spending a year to attain a PhD in attaching boom cranes.

This would require adjusting the crafting, construction, and vehicle modification code quite a bit, but incurring penalties other than lost time or lost materials for failure would be a great upgrade. I think that a permanent maximum durability and accuracy, encumbrance, or armor value penalty to weapons and armor for being poorly made is a great way to reduce the minimum requirements to attempt to make it, and have those penalties cease to exist if an unimpaired person (PC or NPC) of adequate skill makes them. That could also make proficiency relevant. Trying to cook meth without knowing what you’re doing is likely to cause an explosion and fire. 

Construction and vehicle modification below adequate skill level or without necessary proficiency might also have subpar outcomes. 

It would be a massive project to add possible alternate products and mishaps to all of thr crafting, construction, and vehicle modification recipes. Before asking anyone to start on that, I think it’s important to figure out any other changes that should be done at the same time, such as anything to do with tool quality/charges and crafting time (a nailgun and a powered air compressor are much better at driving nails than a rock) or optional components (Jackstands are a lot safer than lifting a car with a scissor jack or your hands, even if you’re strong enough, but someone with a low mechanics might place the jackstands incorrectly and still have the car fall on them). 

I think that it’s not strictly necessary to have the code in place that will use all the mishaps to start writing them, but it will be necessary to have the format fully defined before mishaps can be began. That seems a bit out of scope for this.

**Comment by I-am-Erk** [comment]
There are some things that are probably easier than others. For example, with vehicles, a failure could reduce max durability by a larger amount, or damage other parts in the same tile. Crafts and construction are harder, because we have really no fail states besides 'takes longer' and 'lose ingredients or product', there are multiple issues for that.

In general that's kind of tangential though. I agree it's a good idea to consider a more flexible autolearn system that lets you try more things before you are officially "ready". The new crafting rules we added at the start of H-experimental should handle that very easily. I am not personally sure the best way to do it, though... it's kind of a feature that we're not clogging the crafting menu up at low levels with things that are very hard to craft.

**Comment by Dhanvanthri** [comment]
Literally just bring macros into the game. I know the whole dev team hates it for essentially aesthetic reasons that I understand but can't appreciate even slightly. 
Then you can worry less about clogging menus and more about how to implement the functionality, and macros can serves as the glue that will help transition to better interface over time.

**Comment by I-am-Erk** [comment]
we're getting into the weeds on this one. I added a comment about skill/recipe accessibility to the initial post, and if anyone wants to split off and make a separate issue brainstorming solutions I'll link it.

**Comment by Snaaty** [comment]
I say part way with skills alltogether. They have no actual real-life correlate anyway, as practically any and all examples show. Have only Proficiencies and that's it. 
What is "Fabrication" supposed to be? "Computers"? None of the skills make sense, not to mention nebulous categories such as "devices" which are just hilarious next to ones like "throwing". Obviously they are super deeply ingrained into the game code, so that's probably unfeasable. Skills have always been an incredible abstraction and a crutch from a "realistic" point of view. 

That makes arguments that come from the whole "realism" angle quite unplausible as the fundamental premise of the skill categories that we have in the game are the most un-realistic thing ever, and I don't see them changing, ever. 

My two cents :)

**Comment by ZhilkinSerg** [comment]
> Literally just bring macros into the game.

Which problem do you expect to solve with macros?

**Comment by ZhilkinSerg** [comment]
> I say part way with skills alltogether. They have no actual real-life correlate anyway, as practically any and all examples show. Have only Proficiencies and that's it. What is "Fabrication" supposed to be? "Computers"? None of the skills make sense, not to mention nebulous categories such as "devices" which are just hilarious next to ones like "throwing". Obviously they are super deeply ingrained into the game code, so that's probably unfeasable. Skills have always been an incredible abstraction and a crutch from a "realistic" point of view.
> 
> That makes arguments that come from the whole "realism" angle quite unplausible as the fundamental premise of the skill categories that we have in the game are the most un-realistic thing ever, and I don't see them changing, ever.
> 
> My two cents :)

Skills are a perfectly fine abstraction until someone implements a more nuanced system (like proficiencies and such). Indeed some skills are too generic at the moment, but mostly because this is the legacy code we have. It is not too big, but significant - `538 hits in 48 files`. Something can be easily reworked using proficiencies, but I'm pretty sure we can't replace just everything with proficiences. There are things that a skill dependent which ought to be scalable based on how good character is at something, so we still need something like skill levels (which proficiencies lack).

**Comment by Snaaty** [comment]
Which of the skills do you consider not to be too generic at the moment? 
Skills as per in-game do not reflect real-life abilities at all and proficiencies, theoretical/practical stuff just dance around the issue that it would require a truly enormous amount of ultra specific skills/proficiencies to truly reflect reality.

The problem imho with skills is that they're too generic to make any sense (from a realism pov) and the issue with proficiencies is that they're entirely binary (afaik?) while often requiring significant grind to acquire.

**Comment by I-am-Erk** [comment]
Skills work pretty well for what they represent.

Recipe: I know how to make this specific thing.
Proficiency: I know the techniques and tools used to make a category of things.
Skill: I am good at making things and have a general understanding of how things are made.

Fabrication skill represents the ways a carpenter can more easily pick up blacksmithing. Food handling represents that a sushi chef is not going to have as tough a time learning italian cuisinery as someone who's never chopped up a vegetable.

While these could instead be modeled with (a ton of) new proficiencies, there's not really any reason to, because we have skills for that, and they do a decent job. There are some skills we could debatably do away with in favour of using specific profs instead, but by and large that's not important to this discussion.

**Comment by Raikiri** [comment]
I see people steering away from the actual problems that we're trying to solve, so let me build a little bit of a mind map of what we're doing:
*Problem 1*: It's currently unrealistically easy to achieve mastery in the game.
*Solution to problem 1: As OP proposes, naturally reduce skill gain. Fairly straightforward, but produces two more problems:
*Problem 2*: essential content in the game becomes infeasible to obtain (vehiclework, crafting)
*Problem 3*: it becomes infeasible to interact with the content that people like interacting with (again: mechanics, crafting)

So, to address these new problems, OP proposes to solutions along the lines of "leverage NPC's more". I'm saying "along the lines of", because it obviously includes a lot of foundational work to make NPC's a better part of the game: better AI, possibly direct control, etc etc. But crucially even if NPC's became absolutely perfect overnight, they'd solve *Problem 2*, but only further exacerbate *Probelem 3*, and no amount of balancing, tweaking numbers or foundational improvements can make it any better.

Alternatively, I suggest a solution *along the lines of* making good ingame content not gated behind mastery. I'm saying *along the lines of*, because it can be implemented in a multitude of ways ranging from "a bandaid, but at least it works" to pretty significant reworks. For example, a relatively easy solution is to just re-use our existing capped durability system: instead of gating crafting behind mastery, make amateur crafts have a capped low durability that also comes with lower stats (all melee weapons and armor stats naturally degrade with lower durability). Basically use the current durability cap as a reflection of item's overall quality. At least such solution fixes both *Problem 2* and *Problem 3* at the same time.

**Comment by I-am-Erk** [comment]
I do think that the argument "npc ai isn't great" is worth examining more. What are the things that we're concerned about NPCs helping with where their ai will matter? We have blacksmiths that can craft armour, we can fairly readily add npcs that do the same for tailoring and mechanics. They already do pretty well for construction but this could be examined more closely. Fabrication as well. It's important we work on their combat - and I'm doing what I can there - but I'm not sure I agree with the core idea that their ai is a major barrier. Almost anything you really need should be something you can either get from a static NPC whose AI is largely irrelevant, or at most you've got to get your NPCs into a safe base site and then you don't need to particularly worry if you find them hard to manage in combat. 

In other words, what are the skills you *really must* have at your side at all times, besides combat? Are there really so many that it's going to be a major barrier that forces you to keep a retinue of helpers with you constantly?

**Comment by DeciusBrutus** [comment]
I would say that the biggest thing that NPCs are bad at and hasn’t been filled already is healthcare/wound treatment. 

There’s a lot to unpack there; on the one hand it would be overkill to use an entire dose of antiseptic and a sterile pre-cataclysm bandage on every scratch, but it would be great if they could reasonably patch up themselves and others when appropriate, or at least use their skills and proficiencies when I’m making treatment decisions.

**Comment by Raikiri** [comment]
I think it's fair to keep looking for ways to improve NPC's. However, this thread is about skill learning rate changes, and people will inevitably get salty when NPC's are proposed as a solution to it, no matter how well they function.

**Comment by I-am-Erk** [comment]
I'm fairly confident the same people who consider the idea of helping a mechanic get a shop set up and then getting them to fix a car to your specifications an unacceptable alternative to grinding yourself up to 10 mechanics levels each game are also not going to be satisfied with being told that now they can only make crappy gear. It's still a beneficial part of the puzzle, but I definitely don't think it's the be-all end-all you are imagining.

**Comment by estebandellasilva** [comment]
Another Thing to consider might be some kind of repair Proficency or or a look at the repair requirements - currently equipment can take dmg very easily and this would cause a lot of problems if you cannot repair the equipment and are forced to go to the npc for repairs every 50 zombies you kill (with some equipment you really need high fab to repair)

**Comment by Raikiri** [comment]
Again, if repairing had a chance (based on skill level) to reduce max item durability, the problem vanishes. In real life you can always repair your rapier with tools in your garage, it's just a matter of how usable it'll be afterwards (which again, can be represented with a permanent durability loss).

**Comment by IdleSol** [comment]
@I-am-Erk 
I apologize for the quality of the text. Deepl translate is used.

I don't really like the idea of lowering the skill growth rate. But without specifics and the ability to try these suggestions in game. I can only draw on my own perceptions. Which is pretty unreliable.

It's already been six months. What have you decided to stop at? And if it's not too brazen, in the form of:
- this item we will do by all means
- this item needs to be discussed.
- and this one, we'd like to ask for help with ideas.

>What I'd like to see ideally is the use of avatar age as a 'currency' for this, so the higher skills you choose at start, the older your starting character must be. 

You do have data on how much experience it takes to grow a skill from level 0 to level 10, right? Let it be EXP. And let it take 10 years.

For 1 year, EXP / 10 experience comes out.

What do we need to consider? The age of the character (AGE) and the intelligence of the character (INT). (Think of the formulas as Excel formulas)

Let's take intelligence into account: 
```
(1 + (INT - 8) / 10)
```

Then the character's experience
```
EXP_total = AGE * (EXP / 10 ) * (1 + (INT - 8) /10)
```

Let's make it more complicated: let's take into account childhood (up to 15 years). As a dependence of intelligence on age.
```
INT_current =  INT * min(AGE_current / 15;1)
```
Let's add some nonlinearity:
```
INT_current = roundup(INT * min(AGE_current / 15;1))
```
I don't think we need aging degradation, since the maximum age in the game is 55. But if we do:
```
INT_current = rounddown(INT * max(AGE_current / 55 - 1;0))
```
Total:
```
INT_current = roundup(INT * min(AGE_current / 15;1)) + rounddown(INT * max(AGE_current / 55 - 1;0))
```
Let's add summation in the loop:
```
Fot AGE_current = 6 to AGE do
INT_current = roundup(INT * min(AGE_current / 15;1)) + rounddown(INT * max(AGE_current / 55 - 1;0))
EXP_total =  EXP_total +  (EXP / 10) * (1 + (INT_current - 8) /10)
end
```

We got the experience that the character has accumulated by his age. Add coefficients or change formulas as needed. This is just an example.

Make a change to the character creation.
A profession is not a given skill level, but a given priority of experience distribution.

For example. For the character creation system, there are 10 points worth 10% of experience. Three of them are distributed by the player. The other 7 are distributed according to priority.

Mechanic: driving 1 (10%), mechanics 3 (30%), production 2 (20%), and another 1 (10%) in melee (why not?).

Or don't limit the number of points. But calculate the weight of one point to the total.

**Comment by I-am-Erk** [comment]
IdleSol: Sorry for the slow reply, I was away in february and am going through my issues for updates now.

You're getting a bit ahead of things with your post. We're not close right now to figuring out exactly where skill learning equations should be. First, before that, we need to fix Focus to work much better. Fixing Focus will change all learning speeds already, so we shouldn't guess about how skills need to change until after that is done.

Right now, what we need is mainly to make sure there are other ways to do everything in game besides doing it yourself. I'd like to see NPCs be better at healing and doing first aid, for example, and I'd like to see NPCs that can repair and modify vehicles and equipment like players can. I'd also like NPCs to be able to drive for you. All of those are pretty close right now.

**Comment by DeciusBrutus** [comment]
If the goal of “advancing to 8+ is hard unless you start with 6+” is to be done, I think the only way is to track post-cataclysmic skill gain separately from pre-cataclysm skills. If getting to 8 from 6 is the same amount of hard whether you were a chemistry grad student who started at 6 or a child who started at 1 applied science and then spend a couple months playing with books and science kits now and then, I don’t see how it can be more reasonable for the former than the latter to get the next levels. 

But if the skill gain cost is exponential with number of levels gained since the cataclysm, or perhaps with “self-taught” skill gain (which would make NPC trainers much more relevant), then training from low to high would be more than the sum of training from low to mid and from mid to high.

**Comment by I-am-Erk** [comment]
I don't think skill gain is going to be exponential by level when this is done. Likely each level will be marginally higher than the last, with a kind of sigmoid pattern where the differences are high in the lower level but plateau in the higher, so level 8 and 9 would be pretty similar for example. It's something that would need testing and graphing, but not for a while yet.

But basically the idea is that getting level 1->7 would take quite a lot more XP than getting, like, 7->9 alone. Sorting out the pieces is complicated until we've put down more of the preceding work

**Comment by ZhilkinSerg** [comment]
> track post-cataclysmic skill gain separately from pre-cataclysm skills

There is no reason to do that - blob (or anything else) does not change skill gain

**Comment by DeciusBrutus** [comment]
> > track post-cataclysmic skill gain separately from pre-cataclysm skills
> 
> There is no reason to do that - blob (or anything else) does not change skill gain

This seems empirically false, since a PhD takes more than a month for an average person to get attending school full time, with basic needs and equipment provided for.

**Comment by kevingranade** [comment]
There's no such thing as empirically proving the blob did something. Departures from reality are generally because of implementation details of the game, sometimes arbitrary, and sometimes the result of an inability to match things up with reality due to gameplay considerations.  The only thing that "proves" the blob is causing some effect is if we the designers go "yes I implemented it that way to represent the blob making a thing happen".

**Comment by zachary-kaelan** [comment]
I think a lot of the skills, especially ones focused on crafting, wouldn't fit with such a system.

The cooking skill would be completely incompatible with the system, to the point that it would be almost nonsensical. One doesn't need months of experience with cooking a variety of foods to be able to bake a cake. I don't think there's *any* cooking recipe in the game that the average person would require months of experience to make. One just needs a recipe book and probably some trial and error. The proficiencies system, however, is perfect for this, with the failure chances and taking longer being a really good representation of the problems a newbie would actually face.

For the survival skill, someone definitely doesn't need much experience to be able to spot a glass jar in a bush or to pick berries. While skinning and butchery are definitely a little more difficult, there's still a low upper limit to how much experience you need to get all of the meat and skin from a corpse.

For first aid, there's only so many times you can put antiseptic and bandages on something before you figure out the best way to do it, especially since almost all medical items have directions on the packaging. And there is no learning from experience to do more advanced first aid, because if you require more advanced first aid - say, if you have a punctured lung - then you're dead.

For a lot of things related to chemicals, explosives, and bullets crafting, there are in fact [simple step-by-step guides on how to make a variety of improvised explosives and pipe weapons](https://www.militarynewbie.com/wp-content/uploads/2013/11/TM-21-210-Improvised-Munitions-Handbook-1969-Department-of-the-Army.pdf). 

For something like electronics, there's definitely a lot more experience required to put together something like a smartphone, but also there is no point in working towards crafting the vast majority of electronics as you could just quickly loot them instead.

For tailoring and weapon and armor crafting, again the proficiencies system is perfect. A lot of clothing can definitely just be looted, but the armor and weapons have a lot of variety and specialization.

Mechanics and fabrication are iffy. Construction and vehicle modification definitely shouldn't require specialization to get started with, given how they are core parts of the game that almost everyone will want to explore early and in-depth every playthrough.

For combat skills I'm on board. We just have to make sure that *all* of the combat skills maintain viability into late-game, so that players can't get trapped having mastered something that became useless. That includes throwing, archery, unarmed, launchers, SMGs... pretty much everything other than bashing and rifles.

**Comment by zachary-kaelan** [comment]
> There's no such thing as empirically proving the blob did something.

One thing we can know for certain at least is that the very few humans that are somehow left alive are special.

**Comment by DeciusBrutus** [comment]
> There's no such thing as empirically proving the blob did something. Departures from reality are generally because of implementation details of the game, sometimes arbitrary, and sometimes the result of an inability to match things up with reality due to gameplay considerations. The only thing that "proves" the blob is causing some effect is if we the designers go "yes I implemented it that way to represent the blob making a thing happen".

If a week of full-time practice after the cataclysm represents the same skill growth as a week of full-time practice before, hobbies should be granting lots of related proficiencies. A high school athlete who attends practice twice a week on the practice squad and practices would have maximum athletics and all the proficiencies.

**Comment by I-am-Erk** [comment]
> > There's no such thing as empirically proving the blob did something. Departures from reality are generally because of implementation details of the game, sometimes arbitrary, and sometimes the result of an inability to match things up with reality due to gameplay considerations. The only thing that "proves" the blob is causing some effect is if we the designers go "yes I implemented it that way to represent the blob making a thing happen".
> 
> If a week of full-time practice after the cataclysm represents the same skill growth as a week of full-time practice before, hobbies should be granting lots of related proficiencies. A high school athlete who attends practice twice a week on the practice squad and practices would have maximum athletics and all the proficiencies.

Been a minute since I checked here, sorry.

What Kevin was trying to tell you is that this doesn't mean it is blob magic, it means we need to slow skill gain. Hence the issue post.

**Comment by DeciusBrutus** [comment]
> > > There's no such thing as empirically proving the blob did something. Departures from reality are generally because of implementation details of the game, sometimes arbitrary, and sometimes the result of an inability to match things up with reality due to gameplay considerations. The only thing that "proves" the blob is causing some effect is if we the designers go "yes I implemented it that way to represent the blob making a thing happen".
> > 
> > 
> > If a week of full-time practice after the cataclysm represents the same skill growth as a week of full-time practice before, hobbies should be granting lots of related proficiencies. A high school athlete who attends practice twice a week on the practice squad and practices would have maximum athletics and all the proficiencies.
> 
> Been a minute since I checked here, sorry.
> 
> What Kevin was trying to tell you is that this doesn't mean it is blob magic, it means we need to slow skill gain. Hence the issue post.

If that’s the case and skill gain is dropped to a realistic level, then the gameplay effect is going to be that most skills and proficiencies aren’t learned during the course of a game. 12 hours of expertly guided practice and 8 hours per week of  guided instruction in as close to ideal conditions as possible takes about nine months to develop basic welding proficiency- in that students who complete that amount of training are expected to be able to produce satisfactory welds in the 1g and 2g position as a requirement to advance to the next level of training. 

Even accelerating that timeline by having a more intensive training schedule, that’s spending several months doing very little other than training. Maybe a 20 hour practicum could grant welding familiarity, which could be a hard prerequisite to performing any welding.

**Comment by PatrikLundell** [comment]
Changing learning time to realistic levels currently means most advanced recipes are and will remain unavailable to PCs that don't get the skills at character generation. That would mostly kill the game for a lot of people.

There is a currently theoretical, rather different, game play model that could work around that, which would be the ability to recruit skilled survivors to fill skill gaps. That's currently wholly theoretical, since I'm not aware of any recruitable NPCs that know much about anything, and it also means you'd need to juggle a lot of companions in order to collect a reasonable set of skills at high levels, and the game isn't stellar at companion management.

Note that this doesn't mean any efforts to make it slower to become good at skills will kill the game play, but it can't really be realistic.

**Comment by I-am-Erk** [comment]
This is the usual "verisimilitude vs realism" debate. I don't actually care if we simulate the precise time needed to learn welding. It just has to feel believable and work for play balance. It currently does neither.

**Comment by kevingranade** [comment]
> If that’s the case and skill gain is dropped to a realistic level, then the gameplay effect is going to be that most skills and proficiencies aren’t learned during the course of a game.

Yes that's the point. 

> Changing learning time to realistic levels currently means most advanced recipes are and will remain unavailable to PCs that don't get the skills at character generation.

Yes, making character creation much more meaningful.

> There is a currently theoretical, rather different, game play model that could work around that, which would be the ability to recruit skilled survivors to fill skill gaps.

Yes that's one of the mechanisms this would encourage. The model where the player can fill every role is an antipattern I'd explicitly like to remove.

**Comment by PatrikLundell** [comment]
If you want to change the game from a survivor one where companions are optional, to a management one where you spend your time managing companions, I think you'll have to do things in the opposite order, i.e. FIRST make it viable to recruit skilled companions AND to manage them with a reasonable level of pain, and THEN remove the ability to solo play (unless you spend a lot of time in character creation to acquire every skill and proficiency you'll potentially need).

Note, though, that introducing occasional recruitable skilled NPCs will bring its own pain: incessant job interviews to find the few NPCs that are actually usable (making the trash beggars even more useless), with the associated out of game spreadsheet of filled and missing capabilities. That is, of course, if there even are enough NPCs available, and that's a game setting, so players may inadvertently make this impossible by keeping the number of NPCs down.

It can also be noted that a very common and powerful drive to get people to play games is character development. If there is no meaningful development to be had except equipment, there's a need for a different driver (which likely would attract a different set of people, leaving part of the current base behind). One such driver is exploration, and another one is long term development, neither of which is supported particularly well (the latter is arguably absent, since the lore says things will go completely to shits eventually).

**Comment by I-am-Erk** [comment]
Patrick, most of the nitty gritty of what you just said is in the op, and the rest is you inventing problems that dont exist and only will if we intentionally build them.
